x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -578,6 +578,14 @@ export function parseDataStreamElasticsearchEntry(
     parsedElasticsearchEntry.index_mode = expandedElasticsearch.index_mode;
   }
 
+  if (expandedElasticsearch?.dynamic_dataset) {
```

Comment:
One issue here is that we don't support dotted values in the manifest like `elasticsearch.dynamic_dataset`. We need to be the de-dotting a level higher (will try this in this PR)

Reply:
Do you mean that this setting needs to be nested to work? Like this?
```
elasticsearch:
  dynamic_dataset: true
```

All manifests can be in both formats, is this an specific issue here or also in other places when reading manifests?

Reply:
Yes it was not working for `elasticsearch.*` properties, but I have now fixed this and added tests.